File: lib/bbc/a11y/standards/exactly_one_main_landmark.rb
module BBC
  module A11y
    module Standards
      class ExactlyOneMainLandmark
        def initialize(page)
          @page = page
        end

        def call(errors)
          count = @page.all('*[role=main]').size
          if count != 1
            errors << "A document must have exactly one main landmark." +
                      " Found #{count} elements with role=\"main\"."
          end
        end

      end
    end
  end
end

File: lib/standards/mainLandmark/exactlyOneMainLandmark.js
module.exports = {
  name: 'Exactly one Main landmark',

  test: function ({ $, fail }) {
    var mainLandmarks = $("[role='main']")
    var count = mainLandmarks.length
    if (count === 0) {
      fail('Found 0 elements with role="main".')
    } else if (count > 1) {
      fail('Found ' + count + ' elements with role="main":', mainLandmarks)
    }
  }
}
